index.d.ts

```diff
@@ -0,0 +1,51 @@
+export interface ActionCreator {
+  (...args: any[]): any;
+}
+
+export type Reducer<S> = (state: S, action: any) => S;
+
+export type Dispatch = (action: any) => any;
```

Comment:
It would be great if try to not use too much `any` - it's just like working without support of typscript… :(

Consider more strict approach like:

```
interface Action {
    type: any;
}

interface Reducer<S> extends Function {
    <S>(state: S, action: Action): S;
}

interface ReducersMapObject<S> {
    [key: string]: Reducer<S>
}

interface Dispatch extends Function {
    <A extends Action>(action: A): A;
}
```

Or even better

```
interface Action<T> {
    type: T;
}

interface Reducer<S, T> extends Function {
    <S>(state: S, action: Action<T>): S;
}
…
interface Dispatch<T> extends Function {
    <A extends Action<T>>(action: A): A;
}
```

so anyone can implement his own strict actions:

```
enum ActionType {A, B, C}
interface MyAction implements Redux.Action<ActionType> {
    type: ActionType;
    payload: MyPayloadType;

```

or dynamic ones:

```
interface MyAction implements Redux.Action<any> {}
```